### PR TITLE
FaZe board support

### PR DESCRIPTION
**Pull request by simonguinot**
This PR adds support for the FaZe board which can be found in the Seagate FireCuda Gaming SSD devices. Two chips are embedded on this board: an ASMedia ASM2364 USB-to-PCIe bridge controller and a NXP LPC11U67 MCU. The former is handling the USB type-C to SSD I/Os while the latter is dedicated to the LED effects. This Zephyr port is running on the LPC11U67 MCU.

Note that this PR depends on PR #26101. A  TI LP5030 LED driver is connected the LPC11U67 MCU. PR #26101 provides support for this LED controller and also some LED bindings used in the FaZe DTS as well.

### CODE REVIEW

File: boards/arm/faze/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
++---------+-----------------+----------------------------+
+
+Programming and Debugging
+*************************
+
+Flashing
+========
+
+The NXP LPC11U67 MCU can be flashed by connecting an external debug probe to
+the SWD port (on-board 4-pins J2 header). In the default OpenOCD configuration
```
Comment (dleach02) [inline boards/arm/faze/doc/index.rst:105]: Just curious but is this header actually exposed?

**Review by dleach02 — APPROVED** [review]
LGTM... but need to fix CI failure


/workdir/zephyr/boards/arm/faze/faze.dts:11:10: fatal error: dt-bindings/led/led.h: No such file or directory
  | 11 \| #include <dt-bindings/led/led.h>
  | \|          ^~~~~~~~~~~~~~~~~~~~~~~
  | compilation terminated.
  | CMake Error at /workdir/zephyr/cmake/dts.cmake:153 (message):
  | command failed with return code: 1
  | Call Stack (most recent call first):

File: boards/arm/faze/support/openocd.cfg [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#
+# Seagate FireCuda Gaming SSD (FaZe) board
+#
+
+# An external debug probe must be connected to the SWD port (4-pins J2 header).
+# Here we assume that a ST-LINK in-circuit debugger/programmer is used. You may
+# have to replace it with your own interface here.
+source [find interface/stlink.cfg]
```
Comment (agansari) [inline boards/arm/faze/support/openocd.cfg:8]: why not lpclink?

**Review by agansari — CHANGES_REQUESTED** [review]

File: boards/arm/faze/doc/index.rst (hunk 1 above)
Comment (simonguinot) [reply boards/arm/faze/doc/index.rst:105]: Hi David. No it is not, the casing must be open and the header must be soldered. But it is actually quite easy to do.

**Comment by simonguinot** [comment]
> LGTM... but need to fix CI failure
> 
> /workdir/zephyr/boards/arm/faze/faze.dts:11:10: fatal error: dt-bindings/led/led.h: No such file or directory
>   | 11 | #include <dt-bindings/led/led.h>
>   | | ^~~~~~~~~~~~~~~~~~~~~~~
>   | compilation terminated.
>   | CMake Error at /workdir/zephyr/cmake/dts.cmake:153 (message):
>   | command failed with return code: 1
>   | Call Stack (most recent call first):

Thanks for the approval David. Yes, as stated in the description this PR depends on PR #26101. I am working on it :)

File: boards/arm/faze/support/openocd.cfg (hunk 2 above)
Comment (simonguinot) [reply boards/arm/faze/support/openocd.cfg:8]: Hi Andrei, It is because I am using a ST-LINK probe myself on this board. I'll do the change. Thanks.

**Review by agansari — APPROVED** [review]

**Comment by carlescufi** [comment]
Added a DNM flag until https://github.com/zephyrproject-rtos/zephyr/pull/26101 is merged

**Comment by carlescufi** [comment]
Please rebase now that #26101 is merged.

**Comment by simonguinot** [comment]
> Please rebase now that #26101 is merged.

Done !

**Review by ioannisg — APPROVED** [review]
Board definition looks OK to me

**Pull request merged**